### PR TITLE
[fix bug 1409779] Update Firefox sub navigation for Quantum

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Updates Firefox sub navigation and global nav menu links for Fireofx 57.
- New links are behind the switch `firefox-57-release`.
- Note: there's one new string here `Extensions`, all the others (`Desktop`, `Mobile`, `Blog`) are already in [main.lang](https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/main.lang). 
- For non-English locales, I've made `Extensions` fallback to `Add-ons`. We would still like to translate `Extensions` though, as it's the new term we should use moving forward.
- Note: this PR should not merge until after #5224 so I can update the links to `/firefox/mobile/`.

## Issue / Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1409779

## Testing
Demo: https://www-demo3.allizom.org/en-US/firefox/
- Make sure the new links appear as expected when the switch is enabled, and old content shows when disabled.
- Check for typos?

### CODE REVIEW

File: bedrock/base/templates/includes/global-nav.html [hunk 1]
```diff
@@ -68,6 +68,33 @@ <h3 class="summary" data-id="firefox">
...
               </p>
               <ul class="nav-menu-secondary-links">
+              {% if switch('firefox-57-release') %}
+                <li>
+                  <a href="{{ url('firefox') }}" data-link-type="nav" data-link-name="Desktop" data-link-group="Firefox">
+                    {{ _('Desktop') }}
+                  </a>
+                </li>
+                <li>
+                  <a href="/firefox/mobile/" data-link-type="nav" data-link-name="Mobile" data-link-group="Firefox">
```
Comment (alexgibson) [inline bedrock/base/templates/includes/global-nav.html]: TODO: update link once mobile page is merged.

File: bedrock/firefox/templates/firefox/includes/hub/sub-nav.html [hunk 2]
```diff
@@ -28,11 +35,20 @@
...
-<li><a href="{{ url('firefox.focus') }}" data-link-name="Focus Browser" data-link-type="nav" data-link-position="subnav">{{ focus_text }}</a></li>
-<li><a href="{{ url('firefox.developer') }}" data-link-name="Developer Edition" data-link-type="nav" data-link-position="subnav">{{ _('Developer Edition') }}</a></li>
-<li><a href="{{ url('firefox.features.index') }}" data-link-name="Features" data-link-type="nav" data-link-position="subnav">{{ _('Features') }}</a></li>
-<li><a href="https://support.mozilla.org/?utm_source=www.mozilla.org&amp;utm_medium=referral&amp;utm_campaign=fx-sub-nav" data-link-name="Support" data-link-type="nav" data-link-position="subnav">{{ _('Support') }}</a></li>
+  {% if switch('firefox-57-release') %}
+    {% set extensions_text = _('Add-ons') if not LANG.startswith('en') else _('Extensions') %}
+    <li><a href="{{ url('firefox') }}" data-link-name="Desktop" data-link-type="nav" data-link-position="subnav">{{ _('Desktop') }}</a></li>
+    <li><a href="/firefox/mobile/" data-link-name="Mobile" data-link-type="nav" data-link-position="subnav">{{ _('Mobile') }}</a></li>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/includes/hub/sub-nav.html]: TODO: update link once mobile page is merged.

File: bedrock/base/templates/includes/global-nav.html [hunk 3]
```diff
@@ -68,6 +68,33 @@ <h3 class="summary" data-id="firefox">
                 {{ _('Get the only browser built for people, not profit.') }}
               </p>
               <ul class="nav-menu-secondary-links">
+              {% if switch('firefox-57-release') %}
```
Comment (alexgibson) [inline bedrock/base/templates/includes/global-nav.html]: @pmac this PR adds a switch in the base template, so will be queried on every page. Just checking this isn't going to cause some sort of perf disaster?

Comment (alexgibson) [reply bedrock/base/templates/includes/global-nav.html]: Actually, it looks like we're already using switches in the base templates, so I'm guessing this is fine. Please disregard :)

Comment (pmac) [reply bedrock/base/templates/includes/global-nav.html]: checking switches should be very fast now that they're not in the DB. Thanks for checking though! 💯

**Comment by alexgibson** [comment]
Rebased and added the links to /mobile. this should be good for review & merge when ready.

File: bedrock/firefox/templates/firefox/includes/hub/sub-nav.html [hunk 4]
```diff
@@ -28,11 +35,20 @@
...
-<li><a href="{{ url('firefox.android.index') }}" data-link-name="Android Browser" data-link-type="nav" data-link-position="subnav">{{ android_text }}</a></li>
-<li><a href="{{ url('firefox.ios') }}" data-link-name="iOS Browser" data-link-type="nav" data-link-position="subnav">{{ ios_text }}</a></li>
-<li><a href="{{ url('firefox.focus') }}" data-link-name="Focus Browser" data-link-type="nav" data-link-position="subnav">{{ focus_text }}</a></li>
-<li><a href="{{ url('firefox.developer') }}" data-link-name="Developer Edition" data-link-type="nav" data-link-position="subnav">{{ _('Developer Edition') }}</a></li>
-<li><a href="{{ url('firefox.features.index') }}" data-link-name="Features" data-link-type="nav" data-link-position="subnav">{{ _('Features') }}</a></li>
-<li><a href="https://support.mozilla.org/?utm_source=www.mozilla.org&amp;utm_medium=referral&amp;utm_campaign=fx-sub-nav" data-link-name="Support" data-link-type="nav" data-link-position="subnav">{{ _('Support') }}</a></li>
+  {% if switch('firefox-57-release') %}
+    {% set extensions_text = _('Add-ons') if not LANG.startswith('en') else _('Extensions') %}
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/hub/sub-nav.html]: This is a good question. We can’t use an l10n conditional as it’s a shared .lang file. We did expose strings a while ago though. So perhaps we have enough coverage? /cc @peiying2

**Comment by alexgibson** [comment]
Ok, I'm not going to wait on this. We can always remove the conditional later.

**Pull request merged**